### PR TITLE
added SPACK_ROOT env to MasonEnv

### LINKED ISSUE #15535: Include SPACK_ROOT in mason env
I believe `SPACK_ROOT` should be included in the output of `mason env` since it is an environment variable that impacts mason.

### PR DESCRIPTION
**Pull request by ankingcodes**
This PR adds a feature to MasonEnv to display `SPACK_ROOT` environment variable as it affects Mason. 

fixes #15535

### CODE REVIEW

**Comment by ankingcodes** [comment]
@ben-albrecht Since this was a quick-fix, I went ahead and made a PR for it. 👍

File: tools/mason/MasonEnv.chpl [hunk 1]
```diff
@@ -112,6 +111,14 @@ proc MASON_REGISTRY {
   return registries;
 }
 
+proc getSpackRoot() : string {
```
Comment (Rapiz1) [inline tools/mason/MasonEnv.chpl:114]: What about the name `SPACK_ROOT`? It's consistent with the `MASON_HOME`, `MASON_REGISTRY` precedures.

File: tools/mason/MasonEnv.chpl [hunk 2]
```diff
@@ -112,6 +111,14 @@ proc MASON_REGISTRY {
...
+  const envSpack = getEnv('SPACK_ROOT');
+  const default = getEnv('HOME') + "/.spack";
```
Comment (ankingcodes) [inline tools/mason/MasonEnv.chpl:116]: 'HOME' --> 'MASON_HOME'

File: tools/mason/MasonEnv.chpl (hunk 1 above)
Comment (ankingcodes) [reply tools/mason/MasonEnv.chpl:114]: @Rapiz1  Thanks for the comment, turns out we don't need a function at all.

**Pull request merged**